cypress/tests/ignore-vulnerability.spec.ts

```diff
@@ -0,0 +1,618 @@
+                    cy.findByText(v.name).should('exist');
+                    cy.findByText(riskTextToModifyTo).should('not.exist');
+                  });
+
+                  iVActions.interceptAnalysisItemRes({
+                    idAlias: '@riskyAnalysisId',
+                    intAlias: 'updatedAnalysisItem',
+                  });
+
+                  iVActions.doGoToAnalysisDetailsPage(
+                    '@ignoreVulTestProjectLatestFile',
+                    '@riskyAnalysisId'
+                  );
+
+                  // Open edit analysis drawer
+                  iVActions.openEditDrawer('edit analysis button');
+
+                  // Get updated info of reset analysis
+                  cy.wait('@updatedAnalysisItem')
+                    .its('response.body')
+                    .then((resetAnalysisItem) => {
+                      cy.wrap(resetAnalysisItem).as('resetAnalysisItem');
+                    });
+
+                  // Assert that updated info is being shown correctly
+                  iVActions.getEditAnalysisDrawer().within(() => {
+                    // Edited props should still exist
+                    iVActions.assertEditedAnalysisInfo('@resetAnalysisItem', {
+                      riskTextToModifyTo,
+                      riskyAnalysisRiskText,
+                    });
+
+                    // Reset from project settings alters the "All future analyses" to "Current file only"
+                    iVActions.doResetAnalysis({
+                      resetForCurrentFileOnly,
+                      currentFileOnly: true,
+                      removeOverride: false,
+                    });
+
+                    iVActions.closeEditDrawer();
+                  });
+                });
+            }
+
+            cy.wrap(true).as('testCompleted');
+          }
+        );
+      }
+    );
+  });
+
+  afterEach(() => {
+    // Reset edited analysis if test fails
+    cy.getAliases<boolean>(['@analysisEdited', '@testCompleted']).then(
+      ([analysisEdited, testCompleted]) => {
+        if (analysisEdited && !testCompleted) {
+          cy.window()
+            .its('localStorage')
+            .invoke('getItem', 'ember_simple_auth-session')
+            .then((authInfo) => {
+              const authDetails = authInfo ? JSON.parse(authInfo) : {};
+
+              cy.getAliases<string | AnalysisItemProps>([
+                '@ignoreVulTestProjectLatestFile',
+                '@riskyAnalysis',
+              ]).then(([fileId, analysis]) => {
+                cy.request({
+                  method: 'DELETE',
+                  url: `${API_HOST}/api/files/${fileId}/vulnerability_preferences/${(analysis as AnalysisItemProps).vulnerability}/risk`,
+                  body: { all: true },
+                  headers: {
+                    Authorization: `Basic ${authDetails['authenticated']['b64token']}`,
+                  },
+                });
+              });
+            });
```

Comment:
can we make this a custom command for reuse in other tests

Reply:
Done